### ISSUE TITLE
jsonb support to get read/write jsonb columns to/from play json

### ISSUE THREAD

**Issue by webgress** [mauricio/postgresql-async#134]
Starting a new project and jsonb looks very attractive.
Any pointers on how to deal with jsonb columns ?

**Comment by mauricio** [comment]
Right now there is no support for `jsonb` as including support would require building a jsonb serializer and de-serializer, but we would definitely accept a PR that introduces it :)

**Comment by webgress** [comment]
I'd be happy to, even if I have to hire somebody to do it :)
So two questions:
1 - Is reading/writing jsonb currently possible at all ?
2 - How would I approach "building a jsonb serializer" ? I am not sure how good my scala is, but postgresql with jsonb and non-blocking play is what I want to use for this project.

**Comment by mauricio** [comment]
Right now it will be read as string, not sure how PostgreSQL will produce a string out of it. 

To include it you need to build an encoder and a decoder and register it [here](https://github.com/mauricio/postgresql-async/blob/master/postgresql-async/src/main/scala/com/github/mauricio/async/db/postgresql/column/PostgreSQLColumnEncoderRegistry.scala] and [here](https://github.com/mauricio/postgresql-async/blob/master/postgresql-async/src/main/scala/com/github/mauricio/async/db/postgresql/column/PostgreSQLColumnDecoderRegistry.scala).

**Comment by webgress** [comment]
Great, already something !
So at least I should be able to use play json support to parse/format the strings.
Thank you for a quick reply.

**Comment by TimPigden** [comment]
Hi
@webgress - did you do anything about this? I too want jsonb

**Comment by TimPigden** [comment]
ok - 
So if you convert your json into a string before passing it across to posgres and if you parse it when it comes back there's nothing to do (which mauicio more or less said above). It works already
Thinking about it, there are many different json ASTs in the scala world. Any decision to write a decoder as part of this package would probably be a mistake - as you'd get in the way of the people with a different Json representation. So I'd suggest this doesn't need fixing.

**Comment by mauricio** [comment]
Yeah, which is kind of what I'm thinking as well, we could include a way for you to provide your own JSON parser if you'd like to if people think it's necessary, but I'd rather not force people to use something, specially when there are many different solutions out there already and they aren't even remotely compatible between them.
